### PR TITLE
Basic Dotty + ZIO app

### PR DESCRIPTION
**Pull request by lemastero**
Gitter8 template for simple hello world ZIO application with dotty

### CODE REVIEW

File: src/main/g8/build.sbt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+val zioVersion        = "1.0.0-RC21-2"
```
Comment (molowny) [inline src/main/g8/build.sbt]: what is this indent for?

File: src/main/g8/build.sbt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      List(
+        name := "$name$",
+        organization := "$organization$",
+        version := "0.0.1",
+        scalaVersion := "$dotty_version$"
+      )
+    ),
+    addCompilerPlugin("com.olegpy" %% "better-monadic-for" % "0.3.1"),
+    libraryDependencies ++= Seq(
+      "dev.zio" %% "zio" % zioVersion,
```
Comment (molowny) [inline src/main/g8/build.sbt]: shouldn't it be indented like code below?

Comment (lemastero) [reply src/main/g8/build.sbt]: Yes, it should. Fixed: https://github.com/ScalaConsultants/zio-dotty-quickstart.g8/pull/1/commits/3126fb426e52330eb165581fd0031d3920273760

File: src/main/g8/build.sbt (hunk 1 above)
Comment (lemastero) [reply src/main/g8/build.sbt]: Removed unnecessary indent: https://github.com/ScalaConsultants/zio-dotty-quickstart.g8/pull/1/commits/dd06997cd66e2b2d716c55cc6b678212ca4a82be

**Review by molowny — APPROVED** [review]

File: src/main/g8/src/main/scala/$package$/Main.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import zio.{ App, ExitCode, URIO, ZEnv, ZIO }
+import zio.console.{ getStrLn, putStrLn, Console }
+
+object Main extends App {
+
+  val app: ZIO[Console, IOException, Unit] =
+    for {
+      _    <- putStrLn("What is your name?")
+      name <- getStrLn
+      out  <- putStrLn(s"Hello $name!")
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/Main.scala]: `$` needs to be escaped in templates

File: src/main/g8/src/main/scala/$package$/Main.scala [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.io.IOException
+
+import zio.{ App, ExitCode, URIO, ZEnv, ZIO }
+import zio.console.{ getStrLn, putStrLn, Console }
+
+object Main extends App {
+
+  val app: ZIO[Console, IOException, Unit] =
+    for {
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/Main.scala]: Let's use the new syntax and drop the curly braces

**Review by jczuchnowski — CHANGES_REQUESTED** [review]

File: src/main/g8/src/main/scala/$package$/Main.scala (hunk 3 above)
Comment (lemastero) [reply src/main/g8/src/main/scala/$package$/Main.scala:14]: `sbt g8Test` generate properly template

File: src/main/g8/src/main/scala/$package$/Main.scala (hunk 4 above)
Comment (lemastero) [reply src/main/g8/src/main/scala/$package$/Main.scala]: Removed { in for and Main object

File: src/main/g8/src/main/scala/$package$/Main.scala (hunk 3 above)
Comment (lemastero) [reply src/main/g8/src/main/scala/$package$/Main.scala]: Done: https://github.com/ScalaConsultants/zio-dotty-quickstart.g8/pull/1/commits/0c32a87b4ae6de3d55344d61e05abeac8fc1887b

Looks like `sbt g8Test` do not complain about that :) It is better to test using `g8`

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**